kloppy/domain/models/time.py

```diff
@@ -0,0 +1,262 @@
+    ) -> Union["Time", timedelta]:
+        """
+        Subtract a timedelta or AbsTime from the current AbsTime.
+
+        AbsTime - AbsTime = timedelta
+        AbsTime - timedelta = AbsTime
+
+        The period duration must be taking into account.
+        """
+        if isinstance(other, timedelta):
+            current_timestamp = self.timestamp
+            current_period = self.period
+            while other > current_timestamp:
+                other -= current_timestamp
+                if not current_period.prev_period:
+                    # We reached start of the match, lets just return start itself
+                    return Time(period=current_period, timestamp=timedelta(0))
+
+                current_period = current_period.prev_period
+                current_timestamp = current_period.duration
+
+            return Time(
+                period=current_period, timestamp=current_timestamp - other
+            )
+
+        elif isinstance(other, Time):
+            if self.period >= other.period:
+                diff = self.timestamp
+                current_period = self.period
+                while current_period > other.period:
+                    current_period = current_period.prev_period
+                    diff += current_period.duration
+
+                return diff - other.timestamp
+            else:
+                return -other.__sub__(self)
+        else:
+            raise ValueError(f"Cannot subtract {other}")
+
+    def __add__(self, other: timedelta) -> "Time":
+        assert isinstance(other, timedelta)
+        current_timestamp = self.timestamp
+        current_period = self.period
+        while (current_timestamp + other) > current_period.duration:
+            # Subtract time left in this period
+
+            other -= current_period.duration - current_timestamp
+            if not current_period.next_period:
+                # We reached start of the match, lets just return start itself
+                return Time(
+                    period=current_period, timestamp=current_period.duration
+                )
+
+            current_period = current_period.next_period
+            current_timestamp = timedelta(0)
+
+        return Time(period=current_period, timestamp=current_timestamp + other)
+
+    def __radd__(self, other: timedelta) -> "Time":
+        assert isinstance(other, timedelta)
+        return self.__add__(other)
+
+    def __rsub__(self, other):
+        raise RuntimeError("Doesn't make sense.")
+
+    def __lt__(self, other):
+        return self.period < other.period or (
+            self.period == other.period and self.timestamp < other.timestamp
+        )
+
+    def __str__(self):
+        m, s = divmod(self.timestamp.total_seconds(), 60)
+        return f"P{self.period.id}T{m:02.0f}:{s:02.0f}"
+
+    def __hash__(self):
+        return hash((self.period.id, self.timestamp.total_seconds()))
+
+
+T = TypeVar("T")
+
+
+class TimeContainer(Generic[T]):
```

Comment:
It would be handy to have a `__repr__` method such that you can easily print a player's positions throughout a game.

Reply:
Yes!